source/model.py

```diff
@@ -0,0 +1,159 @@
+import datetime
+import gc
+from typing import List, Mapping
+
+import numpy as np
+import numpy.typing as npt
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+
+
+class Model(nn.Module):
```

Comment:
Why did you define your own model and not use `ResNet18` or similar?

Reply:
Because for a small dataset like this, pretrained complex models are too good and thus it's difficult to observe the improvement.

Reply:
But in the previous version we used an untrained resnet18

Reply:
Of course it works, but in the previous version we did not show how the model improved. Showing improvement is actually not very important (and also trivial because the model trained on more samples surely gets better,) but now that we kind of want to at least showcase this, we need a simple model.

I don't think it's a problem using another model. Here it's just for the convenience of observing performance improvement.

Reply:
Ok :) Is there any reason you have multiple linear layers at then end of the model? It is not very common in conv networks.

Reply:
I wasn't sure if the classifier would be strong enough so I added more layers. Do you think I should change this?

Reply:
I tried this, but the difference is not very clear. I'll keep this simple model.